File: CCC/2018/S1.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class S1 {
    public static void main(String[] args) throws Exception {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(reader.readLine().trim());
        int[] villages = new int[n];
        for (int i = 0; i < n; i++) {
            villages[i] = Integer.parseInt(reader.readLine().trim());
        }

        Arrays.sort(villages);

        double minSize = Integer.MAX_VALUE;
        for (int i = 1; i < n - 1; i++) {
            double villageSize = ((double) villages[i + 1] - (double) villages[i]) / 2f + ((double) villages[i] - (double) villages[i - 1]) / 2f;
            minSize = Math.min(villageSize, minSize);
        }

        System.out.printf("%.1f", minSize);
    }
}

File: CCC/2018/S1.cc
#include <bits/stdc++.h>

using namespace std;

int main(void) {
    int N;
    cin >> N;

    vector<int> villages(N);

    for (int i = 0; i < N; i++) {
        cin >> villages[i];
    }

    sort(villages.begin(), villages.end());

    double minSize = DBL_MAX;

    for (int i = 1; i < N - 1; i++) {
        double villageSize = (double) (villages[i] - villages[i - 1]) / 2 + (double) (villages[i + 1] - villages[i]) / 2;
        minSize = min(minSize, villageSize);
    }

    cout << std::fixed << std::setprecision(1) << minSize << endl;
}